src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -56,6 +57,15 @@ object ConfigConvert extends LowPriorityConfigConvertImplicits {
     override def to(t: T): ConfigValue = ConfigValueFactory.fromAnyRef(t)
   }
 
+  def fromNonEmptyString[T: ClassTag](fromF: String => T): ConfigConvert[T] = new ConfigConvert[T] {
+    lazy val typeName = implicitly[ClassTag[T]]
```

Comment:
In that case, it's better to make the implicit parameter "explicit". See 42ac2de for the change.